digdag-server/src/main/java/io/digdag/server/rs/AttemptResource.java

```diff
@@ -315,7 +315,7 @@ public Response startAttempt(RestSessionAttemptRequest request)
                 // If a group error has occurred,
                 // exclude the group's child/dynamically generated tasks from successTasks
                 // even if they are in SUCCESS state
-                .filter(task -> task.getParentId().isPresent() && !groupRetryErrorTaskIds.contains(task.getParentId().get()))
+                .filter(task -> !groupRetryErrorTaskIds.contains(task.getParentId().orNull()))
```

Comment:
> NullPointerException - if the specified element is null and this list does not permit null elements (optional)

https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/List.html#contains(java.lang.Object)

Whether this method throws NullPointerException depends on the actual class's behavior and it may be a bit naive implementation. How about moving this filter right before `map` below? Like
```
                .filter(task -> !task.getFullName().contains("^sub"))
                .filter(task -> task.getState() == TaskStateCode.SUCCESS)
                .filter(task -> 
                    if (!task.getParentId().isPresent()) {
                        throw new IllegalArgumentException("Resuming successfully completed attempts is not supported");
                    }
                    !groupRetryErrorTaskIds.contains(task.getParentId().get())
                )
                .map(StoredTask::getId)
                      : 
```

Reply:
Thank you for sharing that. I agree with you, and it sounds pretty good! 
I moved this filter based on your review: 4b25b60

Reply:
👍